### DISCUSSION TITLE
fkYAML version 0.3.8

### DISCUSSION THREAD

**Discussion by fktn-k** [fktn-k/fkYAML#352] [Announcements]
### Summary

This release contains the new support of multiline scalars in the deserialization feature.  
As described in the PR #344, the support is limited to single/double quoted styles and multiline scalar keys are being permitted as opposed to the YAML specification.  
The above limitations will, of course, be resolved in future releases.  
Furthermore, a number of bugs in the deserialization feature have been resolved as shown in the Bug Fixes section down below.  
Last but not least, gcc14 and clang18 have been added to the supported compilers list and are now used in the GitHub Actions workflows for testing.

#### :sparkles: New Features

- support flow line folding for single/double quoted scalars by @fktn-k in #344

#### :zap: Improvements

- Normalize newline codes in the input buffer by @fktn-k in #343

#### :bug: Bug Fixes

- Fix parse error on flow containers containing child flow containers by @fktn-k in #345
- Fix parse error on flow container keys by @fktn-k in #346
- Fix wrong parse result from mappings containing the non-first block sequence by @fktn-k in #348 reported by @adjbcbc in #347
- Detect errors on missing flow value separators by @fktn-k in #350
- Fix parse error on empty flow containers by @fktn-k in #351 reported by @adjbcbc in #347

#### :robot: CI

- added gcc14 & clang18 to the compiler version list for testing by @fktn-k in #349

**Full Changelog**: https://github.com/fktn-k/fkYAML/compare/v0.3.7...v0.3.8

<hr /><em>This discussion was created from the release <a href='https://github.com/fktn-k/fkYAML/releases/tag/v0.3.8'>fkYAML version 0.3.8</a>.</em>

**Comment by ARessegetesStery** [comment]
Is it possible to load a yaml array with elements in it having value 0? This version throws exception with msg 
> Floating point value underflow detected.

A minimal example of yaml giving this issue is
```yaml
pos: [0.0, 0.0, 0.0]
```

**Reply by fktn-k** [reply]
@ARessegetesStery   
That should be possible with the following snippet (suppose `n` is a  `fkyaml::node` initialized with 0.0):

```cpp
float f = n.get_value<float>();
// or
double d = n.get_value<double>();
```

The implementation of `get_value()` hasn't been changed for a few releases, if I don't miss something.  
So,  if the conversion fails with the above snippet, it's likely a bug.  
If so, please open an issue, or a PR is also welcome.

**Reply by fktn-k** [reply]
@ARessegetesStery   
Sorry, it's turned out the implementations of `get_value()` for floating point types are wrong.  
So I'll open an issue later and fix it there.

**Reply by ARessegetesStery** [reply]
Sorry I didn't see the reply until now. It seems that the int- and float-versions of `get_value()` does not examine whether the `tmp_int` or `tmp_float` variable is zero. I am submitting a bug report in issues now.

**Reply by fktn-k** [reply]
@ARessegetesStery   
Thank you for the additional investigation!  
I'll check the issue later. (I forgot to open it, and thanks again)

**Reply by fktn-k** [reply]
Addressed as the issue #366, and resolved in the PR #371.
